app/controllers/orders_controller.rb

```diff
@@ -1,35 +1,51 @@
 class OrdersController <ApplicationController
 
   def new
+  end
 
+  def index
+    @orders = current_user.orders
   end
 
   def show
     @order = Order.find(params[:id])
   end
 
   def create
-    order = Order.create(order_params)
-    if order.save
-      cart.items.each do |item,quantity|
-        order.item_orders.create({
-          item: item,
-          quantity: quantity,
-          price: item.price
-          })
-      end
-      session.delete(:cart)
-      redirect_to "/orders/#{order.id}"
-    else
-      flash[:notice] = "Please complete address form to create an order."
-      render :new
-    end
+    order = current_user.orders.create(order_params)
+    order.save ? success(order) : failure
   end
 
-
   private
 
   def order_params
     params.permit(:name, :address, :city, :state, :zip)
   end
+
+  def create_item_orders(order)
+    cart.items.each do |item,quantity|
+      order.item_orders.create({
+        item: item,
+        quantity: quantity,
+        price: item.price
+        })
+      end
+  end
+
+  def successful_redirect
+    session.delete(:cart)
+    flash[:notice] = "Your order has been created"
+    redirect_to "/profile/orders"
+  end
+
+  def success(order)
+    create_item_orders(order)
+    successful_redirect
+  end
```

Comment:
this is the only place where successful_redirect is called, I'm not sure it's necessary. success could have this functionality and still be a small enough chunk of code